.changeset/curvy-mails-retire.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': minor
+---
+
+The TypeScript version has been upgraded to 3.9.7. In this upgrade, the `Action` type was also simplified, and as a result, you should see better TypeScript performance.
```

Comment:
```suggestion
The `Action` type was simplified, and as a result, you should see better TypeScript performance.
```

The TS version upgrade was internal and has no relevance for our consumers, unless we start to require `typescript@^3.9.0`, but based on the changes in this PR - we don't as the changes are rather simple and don't require any new TS features.